src/fidasim.f90

```diff
@@ -2210,6 +2225,294 @@ subroutine make_beam_grid
 
 end subroutine make_beam_grid
 
+subroutine make_pass_grid
+    !+ Makes [[libfida:pass_grid] from user defined inputs
+    integer(HID_T) :: fid, gid
+    integer(HSIZE_T), dimension(2) :: dims
+    logical :: path_valid_spec, path_valid_npa
+
+    real(Float64), dimension(:,:), allocatable :: a_cent, d_cent, phi
+    real(Float64), dimension(:,:), allocatable :: lenses, axes, r0_arr, v0_arr
+    real(Float64), dimension(4,3) :: p_arr
+    real(Float64), dimension(4) :: dt_arr, dt_arr_dum
+    real(Float64), dimension(3) :: vertex111, vertex111_uvw, vertex111_cyl
+    real(Float64), dimension(3) :: vertex221, vertex221_uvw, vertex221_cyl
+    real(Float64), dimension(3) :: n0, nz, cyl_enter, cyl_exit
+    real(Float64), dimension(3) :: r0, v0, p_dum, p, p0, p_enter, p_exit 
+    real(Float64), dimension(3) :: vertex112, vertex112_uvw, vertex112_cyl
+    real(Float64), dimension(3) :: vertex121, vertex121_uvw, vertex121_cyl
+    real(Float64), dimension(3) :: vertex211, vertex211_uvw, vertex211_cyl
+    real(Float64), dimension(3) :: vertex212, vertex212_uvw, vertex212_cyl
+    real(Float64), dimension(3) :: vertex222, vertex222_uvw, vertex222_cyl
+    real(Float64), dimension(3) :: vertex122, vertex122_uvw, vertex122_cyl
+    real(Float64) :: xmin, ymin, xmax, ymax, beam_grid_phimin, beam_grid_phimax
+    real(Float64) :: rmin, zmin, phimin, rmax, zmax, phimax, t
+    integer, dimension(1) :: minpos, maxpos
+    integer :: min_ind, max_ind, npos, cnt, spec_nchan, npa_nchan, i, n, error
+    logical :: outside
+
+    spec_nchan = 0
+    npa_nchan = 0
+    allocate(lenses(3, spec_nchan), axes(3, spec_nchan), &
+             a_cent(3,  npa_nchan), d_cent(3,  npa_nchan))
```

Comment:
I needed to initialize those variables to 0 size in case they're not read in. r0_arr and v0_arr are oriented to have the spectral info before the npa info. If either of the two are not read in, *_nchan = 0 and nothing gets added to r0_arr and v0_arr. Without this, FIDASIM crashes when compiling.